gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -178,6 +237,8 @@ public void onFailure(Throwable throwable) {
           }
         },
         directExecutor());
```

Comment:
@ajaaym, Do you think, we need to perform Callback on the user-provided executor or this is fine?

For a better view:  [ApiFutures.addCallBack(batchResponse...)](https://github.com/rahulKQL/gax-java/blob/c4ed90be86f1624c44c0cbb70af825fd8436b2c1/gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java#L210-L239)